### PR TITLE
Fix explicit composite cache key elements values and improve doc

### LINKED ISSUE #10436: @CacheKey depending on order of properties
**Describe the bug**
A function that invalidates a cache with an `@CacheKey` annotation is only working as expected, if the parameters of the function are in the correct order.

There are two problems here that I can see.
1.  The cache invalidator seems to not ignore the values without a `@CacheKey` annotation for the Key generation
2.  The cache invalidator depends on the order of all `@CacheKey` parameters

**Expected behavior**
All invalidation functions of the TestClass in the Reproducer should actually invalidate the cache for the cached function.
This should not depend on order of parameters, neither if the are non-`@CacheKey` parameters nor if there are only `@CacheKey` parameters (or `@CacheKey` completely missing).

**Actual behavior**
testChangedOrder() and testChangedOrderWithNonCacheKey() are not working because the cache is not invalidated.

**To Reproduce**
I have the following Class inside my Application:
```Java
import io.quarkus.cache.CacheInvalidate;
import io.quarkus.cache.CacheKey;
import io.quarkus.cache.CacheResult;

import javax.inject.Singleton;

@Singleton
public class TestClass {
    public static int callCounter = 0;

    @CacheInvalidateAll(cacheName = "test")
    public void invalidateTestCache() {
        // noop
    }

    @CacheResult(cacheName = "test")
    public String cachedFunction(String randomId, String randomId2) {
        System.out.println("cachedFunction called");
        return randomId + randomId2 + ++callCounter;
    }

    @CacheInvalidate(cacheName = "test")
    public void calculateSomethingRandom(String randomId, String randomId2) {
        // do something randomly
        System.out.println(randomId);
        System.out.println(randomId2);
    }

    @CacheInvalidate(cacheName = "test")
    public void calculateSomeOtherRandom(@CacheKey String randomId, @CacheKey String randomId2, String param1) {
        // do something randomly
        System.out.println(param1);
        System.out.println(randomId);
        System.out.println(randomId2);
    }

    @CacheInvalidate(cacheName = "test")
    public void calculateSomeOtherRandomChangedOrder(@CacheKey String randomId2, @CacheKey String randomId, String param1) {
        // do something randomly
        System.out.println(param1);
        System.out.println(randomId);
        System.out.println(randomId2);
    }

    @CacheInvalidate(cacheName = "test")
    public void calculateSomeOtherRandomWithOtherOrderAndNonCacheKey(String param1, @CacheKey String randomId, @CacheKey String randomId2) {
        // do something randomly
        System.out.println(param1);
        System.out.println(randomId);
        System.out.println(randomId2);
    }
}
```
And the following test:
```Java
import io.quarkus.test.junit.QuarkusTest;
import org.junit.jupiter.api.Test;

import javax.inject.Inject;

import static org.junit.jupiter.api.Assertions.assertEquals;

@QuarkusTest
public class TempTest {

    @Inject
    public TestClass testClass;

    @BeforeEach
    public void reset() {
        TestClass.callCounter = 0;
        testClass.invalidateTestCache();
    }

    @Test
    public void testCache() {
        assertEquals("testtest21", testClass.cachedFunction("test", "test2"));
        assertEquals("testtest21", testClass.cachedFunction("test", "test2"));

        testClass.calculateSomethingRandom("test", "test2");
        assertEquals("testtest22", testClass.cachedFunction("test", "test2"));
        assertEquals("testtest22", testClass.cachedFunction("test", "test2"));

        testClass.calculateSomeOtherRandom("test", "test2", "randomString");
        assertEquals("testtest23", testClass.cachedFunction("test", "test2"));
        assertEquals("testtest23", testClass.cachedFunction("test", "test2"));
    }

    @Test
    public void testChangedOrder() {
        assertEquals("testtest21", testClass.cachedFunction("test", "test2"));
        assertEquals("testtest21", testClass.cachedFunction("test", "test2"));

        testClass.calculateSomeOtherRandomChangedOrder("test2", "test", "randomString");
        assertEquals("testtest22", testClass.cachedFunction("test", "test2"));
        assertEquals("testtest22", testClass.cachedFunction("test", "test2"));
    }

    @Test
    public void testChangedOrderWithNonCacheKey() {
        assertEquals("testtest21", testClass.cachedFunction("test", "test2"));
        assertEquals("testtest21", testClass.cachedFunction("test", "test2"));

        testClass.calculateSomeOtherRandomWithOtherOrderAndNonCacheKey("randomString", "test", "test2");
        assertEquals("testtest22", testClass.cachedFunction("test", "test2"));
        assertEquals("testtest22", testClass.cachedFunction("test", "test2"));
    }

}
```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)

**Environment (please complete the following information):**
 - Output of `java -version`:  
```
openjdk 11.0.2 2019-01-15
OpenJDK Runtime Environment 18.9 (build 11.0.2+9)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.2+9, mixed mode)
```
 - Quarkus version or git rev: `1.5.2.Final`
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
```
Java version: 11.0.2, vendor: Oracle Corporation, runtime: ...
Default locale: de_DE, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"
```

### PR DESCRIPTION
**Pull request by gwenneg**
Fixes #10436.

cc @gsmet @geoand

### CODE REVIEW

File: docs/src/main/asciidoc/cache.adoc [hunk 1]
```diff
@@ -327,6 +327,49 @@ method annotated with `@CacheResult` or `@CacheInvalidate`.
 
 This annotation is optional and should only be used when some of the method arguments are NOT part of the cache key.
 
+=== Composite cache key building logic
+
+When a cache key is built from several method arguments, whether they are explicitly identified with `@CacheKey` or not, the building logic depends on these arguments order in the method signature. On the other hand, the arguments names are not used at all and do not have any effect on the cache key.
```
Comment (geoand) [inline docs/src/main/asciidoc/cache.adoc:332]:
```suggestion
When a cache key is built from several method arguments, whether they are explicitly identified with `@CacheKey` or not, the building logic depends on the order of these arguments in the method signature. On the other hand, the arguments names are not used at all and do not have any effect on the cache key.
```

**Review by geoand — APPROVED** [review]
Thanks!

I added a small doc suggestion

**Comment by gwenneg** [comment]
Suggestion applied, thanks for the review @geoand!

**Review by gsmet — APPROVED** [review]
Nice work, thanks!

**Pull request merged**